### PR TITLE
[ASL-4411] Move project condition mapping to asl-metrics, add tests

### PR DESCRIPTION
**Pull request by jeff-horton-ho-sas**
* Move the additional mapping that asl-internal-ui does for the ppl-conditions report into asl-metrics to match the standard for other reports
* Include update from ASL-4411 so that inspector added conditions show their title in the condition column (Supercedes https://github.com/UKHomeOffice/asl-internal-ui/pull/492)
* Add tests

### CODE REVIEW

**Review by parvinba — APPROVED** [review]

File: lib/reports/ppl-conditions/parse.js [hunk 1]
```diff
@@ -35,12 +36,59 @@ const parse = project => {
...
   }
+
+  const defaultCondition = get(CONDITIONS_SPEC[condition.level], condition.path) || {};
+  const isCustom = condition.custom || condition.key === 'custom';
+  const isRA = condition.key === 'retrospective-assessment';
+
+  const requiresEditing = isCustom || isRA ? false : ((defaultCondition.requiresEditing && !condition.edited) || false);
+
+  return {
+    ...pick(condition, 'establishment', 'licence_number', 'title', 'status', 'schema_version', 'issue_date', 'level', 'protocol_name', 'type'),
```
Comment (farrukhishaq) [inline lib/reports/ppl-conditions/parse.js:82]: nice replacement from omit.

File: lib/reports/ppl-conditions/parse.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
-const { omit, concat, flatten, pick } = require('lodash');
+const { omit, get, concat, pick } = require('lodash');
 const moment = require('moment');
+const CONDITIONS_SPEC = require('@asl/projects/client/constants/conditions').default;
+const RA = require('@asl/projects/client/constants/retrospective-assessment').default;
 
 const getCondition = (condition, level, protocolTitle = '') => {
   return {
```
Comment (farrukhishaq) [inline lib/reports/ppl-conditions/parse.js:7]: in the original code it was 

nst defaults = {
   level: '',
   protocol_name: '',
   type: '',
   condition: 'none',
   requires_editing: '',
   edited: '',
   content: ''
 };
 
 whereas yours is only a subset?

**Comment by farrukhishaq** [comment]
Good refactoring especially the confidence in adding test. I have one question above other than that, I am happy to approve.

File: lib/reports/ppl-conditions/parse.js (hunk 2 above)
Comment (jeff-horton-ho-sas) [reply lib/reports/ppl-conditions/parse.js:7]: This is the "getCondition" function that was originally in this file that normalises the conditions from the project/protocol data. See "conditionToReportRow" for the mapper from asl-internal-ui - that should have the same behaviour, except it's not including some properties that was being implicitly ignored before due to the way that the CSV mapper uses the first entry to determine the headers (hence omit -> pick).

I will rename this so it's clearer what it's for though.

**Pull request merged**